sympy/integrals/integrals.py

```diff
@@ -397,6 +397,13 @@ def doit(self, **hints):
         if not hints.get('integrals', True):
             return self
 
+        # fixes issue 20693
+        from sympy import Function, Pow, symbols
+        if isinstance(self.args[1][0], Function):
+            if isinstance(self.args[0], Pow):
+                t = symbols('t')
+                return self.subs(self.args[1][0], t).doit(**hints).subs(t, self.args[1][0])
+
```

Comment:
On [this branch](https://github.com/hyadav2k/sympy/tree/20693_integrate_func),
```python
IPython console for SymPy 1.8.dev (Python 3.6.9-64-bit) (ground types: python)

These commands were executed:
>>> from __future__ import division
>>> from sympy import *
>>> x, y, z, t = symbols('x y z t')
>>> k, m, n = symbols('k m n', integer=True)
>>> f, g, h = symbols('f g h', cls=Function)
>>> init_printing()

Documentation can be found at https://docs.sympy.org/dev


In [1]: f = Function('f')

In [2]: Integral(sin(f(x)), f(x))
Out[2]: 
⌠                  
⎮ sin(f(x)) d(f(x))
⌡                  

In [3]: _.doit()
Out[3]: f(x)⋅sin(f(x))
```
On Wolfram Alpha,
https://www.wolframalpha.com/input/?i=integrate+sin%28f%28x%29%29d%28f%28x%29%29%2Fdx

May be the root cause is somewhere inside the algorithm of integral computation.

Reply:
Thank you for reviewing, tried to fix it by directly checking for the case of integration w.r.t function and directly applying subs.
Before, it was confined to only objects of type `Pow`, but now it may work for other cases.
Kindly share your opinion on it.
Thanks